build-locally.sh

```diff
@@ -0,0 +1,127 @@
+}
+debug() { printf "${white}%s${reset}\n" "$@"
+}
+info() { printf "${white}➜ %s${reset}\n" "$@"
+}
+success() { printf "${green}✔ %s${reset}\n" "$@"
+}
+error() { printf "${red}✖ %s${reset}\n" "$@"
+}
+warn() { printf "${tan}➜ %s${reset}\n" "$@"
+}
+bold() { printf "${bold}%s${reset}\n" "$@"
+}
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@"
+}
+
+
+#--------------------------------------------------------------------------
+# 
+# Main script logic
+#
+#--------------------------------------------------------------------------
+
+
+#--------------------------------------------------------------------------
+# Check that the ../framework is present.
+h2 "Making sure the openapi yaml file is available..."
+if [[ ! -e "../framework" ]]; then
+    error "../framework is not present. Clone the framework repository."
+    info "The openapi.yaml file from the framework repository is needed to generate a go client for the rest API"
+    exit 1
+fi
+
+if [[ ! -e "../framework/openapi.yaml" ]]; then 
+    error "File ../framework/openapi.yaml is not found."
+    info "The openapi.yaml file from the framework repository is needed to generate a go client for the rest API"
+    exit 1
+fi
+success "OK"
+
+#--------------------------------------------------------------------------
+# Create a temporary folder which is never checked in.
+h2 "Making sure the tools folder is present."
+mkdir -p tools
+rc=$? ; if [[ "${rc}" != "0" ]]; then error "Failed to ensure the tools folder is present. rc=${rc}" ; exit 1 ; fi
+success "OK"
+
+#--------------------------------------------------------------------------
+# Download the open api generator tool if we've not got it already.
+export OPENAPI_GENERATOR_CLI_VERSION="6.2.0"
+export OPENAPI_GENERATOR_CLI_JAR=${BASEDIR}/tools/openapi-generator-cli.jar
+if [[ ! -e ${OPENAPI_GENERATOR_CLI_JAR} ]]; then
+    info "The openapi generator tool is not available, so download it."
+    export OPENAPI_GENERATOR_CLI_SITE="https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli"
+    wget ${OPENAPI_GENERATOR_CLI_SITE}/${OPENAPI_GENERATOR_CLI_VERSION}/openapi-generator-cli-${OPENAPI_GENERATOR_CLI_VERSION}.jar \
+    -O ${OPENAPI_GENERATOR_CLI_JAR}
+    rc=$? ; if [[ "${rc}" != "0" ]]; then error "Failed to download the open api generator tool. rc=${rc}" ; exit 1 ; fi
+    success "Downloaded OK"
+fi
+
+#--------------------------------------------------------------------------
+# Invoke the generator
+h2 "Generate the openapi client go code..."
+./genapi.sh 2>&1 > tools/generate-log.txt
+rc=$? ; if [[ "${rc}" != "0" ]]; then cat tools/generate-log.txt ; error "Failed to generate the code from the yaml file. rc=${rc}" ; exit 1 ; fi
+rm -f tools/generate-log.txt
+success "Code generation OK"
+
+#--------------------------------------------------------------------------
+# Invoke the generator again with different parameters
+h2 "Generate the openapi client go code... part II"
```

Comment:
Why do we generate the client code twice ?
- Once with (genapi.sh)
```
java -jar ${OPENAPI_GENERATOR_CLI_JAR} generate \
-i ${OPENAPI_YAML_FILE} \
-g go \
-o pkg/galasaapi \
--additional-properties=packageName=galasaapi \
--additional-properties=isGoSubmodule=true
```
and once with (generate.sh)
```
java -jar ${OPENAPI_GENERATOR_CLI_JAR} generate \
-i ${OPENAPI_YAML_FILE} \
-g go \
-o pkg/galasaapi \
--additional-properties=packageName=galasaapi
```

Reply:
As of why, I dont know. The difference being if the generated code is a standalone go module, or if it is a submodule to be added to the pkg/ directory. From the small context i can see, Im not sure why we do the non submodule component as this is fitting into the CLI module